aws/container-linux/kubernetes/workers/workers.tf

```diff
@@ -26,11 +26,17 @@ resource "aws_autoscaling_group" "workers" {
     create_before_destroy = true
   }
 
-  tags = [{
+  tags {
     key                 = "Name"
     value               = "${var.name}-worker"
     propagate_at_launch = true
-  }]
+  }
+
+  tags {
+    key                 = "kubernetes.io/cluster/${var.cluster_name}"
+    value               = "owned"
+    propagate_at_launch = true
```

Comment:
Similarly, the same tag for `controllers` should have `propagate_at_launch = true` as well since we want the tag to be on every instance as well

Reply:
That only applies when you're tagging launch configurations, it's a special tag syntax. The tags can apply to just the launch config or to instances launched from it.

Reply:
@bendrucker I did not realize that `controllers` are not under an autoscaling group. Should they?

Reply:
Controllers run etcd which requires consensus among nodes. Being in an autoscaling group creates all kinds of weird problems. You need a discovery mechanism because statically defining a list of peer IPs is no longer possible. If instances are terminated due to failed checks you may end up making your cluster permanently inoperable. If that instance had come back, everything would have been fine.

ASGs are good for stateless apps but are hard to adapt to stateful apps. Auto-healing for etcd would be great but it has to be tightly controlled/integrated. Every node would need to report it's status and autoscaling would have to carefully schedule terminations based on cluster state. You could get really advanced with CloudWatch metrics and alarms to try to do it properly but it's never something I'd expect of Typhoon. Super hard to replicate across clouds.

If a master node fails we can taint and replace it via terraform.